src/sumcheck/sumcheck_single.rs

```diff
@@ -250,6 +250,27 @@ where
         self.sum = combination_randomness * sumcheck_poly.evaluate_at_point(folding_randomness);
     }
 
+    /// Compresses the polynomial and weight evaluations by reducing the number of variables.
+    ///
+    /// Given a multilinear polynomial `p(X1, ..., Xn)`, this function eliminates `X1` using the
+    /// folding randomness `r`:
+    /// \begin{equation}
+    ///     p'(X_2, ..., X_n) = (p(1, X_2, ..., X_n) - p(0, X_2, ..., X_n)) \cdot r
+    ///     + p(0, X_2, ...,X_n)
```

Comment:
In general, this maps p(X_1, ..., X_n) to p(r, ..., X_n). So simply fixes the first randomness. 
The equation is p'(X_2, ..., X_n) = (1- r) * p(0, X_2, ..., X_n) + r * p(1, X_2, ..., X_n)